tempfile.go

```diff
@@ -4,33 +4,40 @@ import (
 	"bufio"
 	"encoding/binary"
 	"io"
-	"io/ioutil"
 	"os"
 )
 
 type tempWriter struct {
-	f *os.File
-	c compressedWriter
-	w *bufio.Writer
+	f        *os.File
+	c        compressedWriter
+	w        *bufio.Writer
+	keepFile bool
 
 	scratch []byte
 	offsets []int64
 	size    int64
 }
 
-func newTempWriter(dir string, compress Compression) (*tempWriter, error) {
-	f, err := ioutil.TempFile(dir, "extsort")
+func newTempWriter(dir string, compress Compression, keepFile bool) (*tempWriter, error) {
+	f, err := os.CreateTemp(dir, "extsort")
 	if err != nil {
 		return nil, err
 	}
+	if !keepFile {
+		// immediately remove for less chance of orphaning.
```

Comment:
I don't think this is valid. Deleting open files may work on some file systems but it's not generally portable.

Reply:
True, I can add the build tags to make it only darwin plus linux.

Reply:
There is https://github.com/tmthrgd/tmpfile which would be a little better, but doesn't support the `pattern` at the moment, so what I have in here I think is simple enough.